lib/Service/RoomService.php

```diff
@@ -543,6 +560,45 @@ public function setType(Room $room, int $newType, bool $allowSwitchingOneToOne =
 		$this->dispatcher->dispatchTyped($event);
 	}
 
+	/**
+	 * @param Room $room
+	 * @param int $newType Currently it is only allowed to change between `Room::TYPE_GROUP` and `Room::TYPE_PUBLIC`
+	 * @param bool $allowSwitchingOneToOne Allows additionally to change the type from `Room::TYPE_ONE_TO_ONE` to `Room::TYPE_ONE_TO_ONE_FORMER`
+	 * @throws TypeException
+	 */
+	public function makePublic(Room $room, string $password = ''): void {
+		if ($room->getType() === Room::TYPE_PUBLIC) {
+			return;
+		}
+
+		if ($room->getType() !== Room::TYPE_GROUP) {
+			throw new TypeException(TypeException::REASON_TYPE);
+		}
+
+		if ($password === '') {
+			throw new PasswordException(PasswordException::REASON_VALUE, $this->l10n->t('Password needs to be set'));
+		}
```

Comment:
Only when the config is enforced?

Reply:
see line https://github.com/nextcloud/spreed/blob/d2bd3e03ad5819b8809339e6d5c4db95ec6a1fd7/lib/Controller/RoomController.php#L1427C2-L1435C3